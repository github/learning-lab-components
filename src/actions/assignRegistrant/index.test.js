const assignRegistrant = require('./')
const mockContext = require('../../tests/mockContext')

describe('assignRegistrant', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      issue: {
        number: 1
      }
    }, {
      request: jest.fn(),
      search: {
        issues: jest.fn(() => Promise.resolve({
          data: {
            items: [{ title: 'My issue', number: 2 }]
          }
        }))
      }
    })
  })

  it('closes the issue from the webhook payload', async () => {
    await assignRegistrant(context, {})
    expect(context.github.request).toHaveBeenCalled()
    expect(context.github.request.mock.calls).toMatchSnapshot()
  })

  it('closes an issue with an issue string', async () => {
    await assignRegistrant(context, { issue: 'My issue' })
    expect(context.github.request).toHaveBeenCalled()
    expect(context.github.request.mock.calls).toMatchSnapshot()
  })

  it('closes an issue with an issue number', async () => {
    await assignRegistrant(context, { issue: 2 })
    expect(context.github.request).toHaveBeenCalled()
    expect(context.github.request.mock.calls).toMatchSnapshot()
  })

  it('throws an error if the issue cannot be found', async () => {
    try {
      await assignRegistrant(context, { issue: 'My issue' })
    } catch (err) {
      expect(err).toMatchSnapshot()
    }
  })
})
