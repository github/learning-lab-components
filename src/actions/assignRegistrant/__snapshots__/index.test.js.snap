// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assignRegistrant closes an issue with an issue number 1`] = `
Array [
  Array [
    Object {
      "assignees": Array [
        "JasonEtco",
      ],
      "headers": Object {
        "accept": "application/vnd.github.symmetra-preview+json",
      },
      "method": "POST",
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
      "url": "/repos/:owner/:repo/issues/:number/assignees",
    },
  ],
]
`;

exports[`assignRegistrant closes an issue with an issue string 1`] = `
Array [
  Array [
    Object {
      "assignees": Array [
        "JasonEtco",
      ],
      "headers": Object {
        "accept": "application/vnd.github.symmetra-preview+json",
      },
      "method": "POST",
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
      "url": "/repos/:owner/:repo/issues/:number/assignees",
    },
  ],
]
`;

exports[`assignRegistrant closes the issue from the webhook payload 1`] = `
Array [
  Array [
    Object {
      "assignees": Array [
        "JasonEtco",
      ],
      "headers": Object {
        "accept": "application/vnd.github.symmetra-preview+json",
      },
      "method": "POST",
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
      "url": "/repos/:owner/:repo/issues/:number/assignees",
    },
  ],
]
`;
